subsys/net/lib/ftp_client/src/ftp_client.c

```diff
@@ -768,7 +767,7 @@ int ftp_get(const char *file)
 	return ret;
 }
 
-int ftp_put(const char *file, const uint8_t *data, uint16_t length, int type)
```

Comment:
With reference to the ftp put data length, my understanding is that send (line 236 & 203) returns the number of bytes transmitted, hence will loop in 4K chunks until all data in the buffer are sent. I do not think that increasing the maximum length will require changes to the upstream data mode, I could be mistaken however. Otherwise is there a possible workaround for sending large quantities of data over FTP? 

I am not too familiar with the pull request process, so please let me know if I am missing something.

Reply:
I revoked the comment. The ftp_client lib was prepared for SLM FTP fucntion, which receive uplink data from MCU by data mode, and which in turn cannot be more than 4K. If a SoC application wants to send more than 64K (max uint_16) at once, you'll need a buffer of 64K first.

The streaming socket interface has removed the 4K limitaton as it supports larger size, but there are still limitation on modem side (TCP buffer size). It's highly recommended to send small-size chunk of data per packet over socket, especially for NB-IoT environment.

Reply:
I see, thank you for the more detailed explanation.  So for sending more than 64K of data from an SoC application I should use multiple calls to ftp_put()?

Reply:
Yes and recommed to follow the data mode model i.e. less than 4K each time.
You could also tune uplink sending performance by adjusting your buffer size.